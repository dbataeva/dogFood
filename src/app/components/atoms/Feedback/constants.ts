export const SNACK_BAR_VISIBILITY_TIME = 3000;
