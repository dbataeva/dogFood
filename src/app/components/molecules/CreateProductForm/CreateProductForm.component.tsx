import CardActions from '@mui/material/CardActions';
import { FC, memo } from 'react';
import { Box, Button, TextField } from '@mui/material';
import { Controller, SubmitHandler, useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';

import {
	TEXT_MAP,
	createProductScheme,
	CREATE_PRODUCT_FORM_TEST_ID_MAP,
} from './constants';
import {
	AddNewProductRequest,
	ErrorResponse,
	useAddProductMutation,
} from '../../../../api';
import { Feedback } from '../../atoms';
import { useFeedback } from '../../../hooks';

export const CreateProductForm: FC = memo(() => {
	const {
		feedbackVisibility,
		setFeedbackVisibility,
		feedbackMessage,
		feedbackState,
		showSuccessfulFeedback,
		showFailureMessage,
	} = useFeedback();

	const {
		control,
		handleSubmit,
		formState: { errors, isSubmitting },
	} = useForm<AddNewProductRequest>({
		defaultValues: {
			name: '',
			pictures: '',
			description: '',
		},
		resolver: yupResolver(createProductScheme),
	});

	const [addProductFn] = useAddProductMutation();

	const submitHandler: SubmitHandler<AddNewProductRequest> = async (data) => {
		try {
			await addProductFn(data).unwrap();
			showSuccessfulFeedback(TEXT_MAP.successfulAdding);
		} catch (error) {
			showFailureMessage((error as ErrorResponse).data.message);
		}
	};

	return (
		<>
			<Box
				noValidate
				component='form'
				sx={{ width: 400 }}
				data-testid={CREATE_PRODUCT_FORM_TEST_ID_MAP.form}
				onSubmit={handleSubmit(submitHandler)}>
				<Controller
					name='name'
					control={control}
					render={({ field }) => (
						<TextField
							required
							fullWidth
							margin='normal'
							error={!!errors.name}
							data-testid={CREATE_PRODUCT_FORM_TEST_ID_MAP.nameField}
							label={!!errors.name ? TEXT_MAP.nameError : TEXT_MAP.name}
							{...field}
						/>
					)}
				/>
				<Controller
					name='price'
					control={control}
					render={({ field }) => (
						<TextField
							required
							fullWidth
							margin='normal'
							error={!!errors.price}
							data-testid={CREATE_PRODUCT_FORM_TEST_ID_MAP.priceField}
							label={!!errors.price ? TEXT_MAP.priceError : TEXT_MAP.price}
							{...field}
						/>
					)}
				/>
				<Controller
					name='discount'
					control={control}
					render={({ field }) => (
						<TextField
							required
							fullWidth
							margin='normal'
							error={!!errors.discount}
							data-testid={CREATE_PRODUCT_FORM_TEST_ID_MAP.discountField}
							label={
								!!errors.discount ? TEXT_MAP.discountError : TEXT_MAP.discount
							}
							{...field}
						/>
					)}
				/>
				<Controller
					name='pictures'
					control={control}
					render={({ field }) => (
						<TextField
							required
							fullWidth
							margin='normal'
							error={!!errors.pictures}
							data-testid={CREATE_PRODUCT_FORM_TEST_ID_MAP.picturesField}
							label={
								!!errors.pictures ? TEXT_MAP.picturesError : TEXT_MAP.pictures
							}
							{...field}
						/>
					)}
				/>
				<Controller
					name='stock'
					control={control}
					render={({ field }) => (
						<TextField
							required
							fullWidth
							margin='normal'
							error={!!errors.stock}
							data-testid={CREATE_PRODUCT_FORM_TEST_ID_MAP.stockField}
							label={!!errors.stock ? TEXT_MAP.stockError : TEXT_MAP.stock}
							{...field}
						/>
					)}
				/>
				<Controller
					name='description'
					control={control}
					render={({ field }) => (
						<TextField
							required
							fullWidth
							margin='normal'
							error={!!errors.description}
							data-testid={CREATE_PRODUCT_FORM_TEST_ID_MAP.descriptionField}
							label={
								!!errors.description
									? TEXT_MAP.descriptionError
									: TEXT_MAP.description
							}
							{...field}
						/>
					)}
				/>
				<CardActions disableSpacing>
					<Button
						type='submit'
						variant='text'
						disabled={isSubmitting}
						data-testid={CREATE_PRODUCT_FORM_TEST_ID_MAP.submitButton}>
						{TEXT_MAP.finish}
					</Button>
				</CardActions>
			</Box>
			<Feedback
				state={feedbackState}
				message={feedbackMessage}
				isVisible={feedbackVisibility}
				setIsVisible={setFeedbackVisibility}
			/>
		</>
	);
});

CreateProductForm.displayName = 'CreateProductForm';
