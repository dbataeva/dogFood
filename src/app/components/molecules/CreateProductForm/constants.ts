import { number, object, string } from 'yup';

export const TEXT_MAP = {
	name: 'Наименование:',
	price: 'Цена:',
	discount: 'Скидка:',
	pictures: 'Ссылка на изображение:',
	stock: 'Количество в наличии:',
	description: 'Описание товара:',
	finish: 'Добавить продукт',
	successfulAdding: 'Продукт успешно добавлен',
	nameError: 'Имя должно быть заполнено',
	priceError: 'Цена должна быть заполнена',
	discountError: 'Скидка должна быть заполнена',
	picturesError: 'Добавьте ссылку на изобржения продукта',
	stockError: 'Количество в наличии должно быть заполнено',
	descriptionError: 'Добавьте описание продукта',
};

export const createProductScheme = object({
	name: string().required(),
	price: number().required(),
	discount: number().required(),
	pictures: string().url().required(),
	stock: number().required(),
	description: string().required(),
});

export const TEST_ID_MAP = {
	form: 'form',
	nameField: 'nameField',
	priceField: 'priceField',
	discountField: 'discountField',
	picturesField: 'picturesField',
	stockField: 'stockField',
	descriptionField: 'descriptionField',
	submitButton: 'submitButton',
};
