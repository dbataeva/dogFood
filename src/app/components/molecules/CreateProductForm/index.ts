export { CreateProductForm } from './CreateProductForm.component';
