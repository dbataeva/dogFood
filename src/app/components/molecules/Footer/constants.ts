export const TEXT_MAP = {
	catalog: 'Каталог',
	sale: 'Акции',
	news: 'Новости',
	review: 'Отзывы',
	paymentAndDelivery: 'Оплата и доставка',
	fac: 'Часто спрашивают',
	feedback: 'Обратная связь',
	contacts: 'Контакты',
};

export const LEFT_MENU_ITEMS = [
	TEXT_MAP.catalog,
	TEXT_MAP.sale,
	TEXT_MAP.news,
	TEXT_MAP.review,
];

export const RIGHT_MENU_ITEMS = [
	TEXT_MAP.paymentAndDelivery,
	TEXT_MAP.fac,
	TEXT_MAP.feedback,
	TEXT_MAP.contacts,
];

export const TEST_ID_MAP = {
	footer: 'footer',
	logo: 'footerLogo',
	contacts: 'contacts',
};
