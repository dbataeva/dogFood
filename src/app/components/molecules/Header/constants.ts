export const MOBILE_MENU_ID = 'primary-search-account-menu-mobile';

export const HEADER_TEST_ID_MAP = {
	header: 'header',
	logo: 'headerLogo',
	searchBar: 'searchBar',
	navIcons: 'navIcons',
};
