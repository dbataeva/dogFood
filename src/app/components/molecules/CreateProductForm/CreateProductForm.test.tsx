import { screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { userEvent } from '@testing-library/user-event';
import ReactTestRenderer from 'react-test-renderer';

import * as productsApi from '../../../../api/productsApi';
import { CREATE_PRODUCT_FORM_TEST_ID_MAP } from './constants';
import { testProvidersFn } from '../../../testProviders';
import * as appStore from '../../../store/appStore';
import { CreateProductForm } from './CreateProductForm.component';

const correctPictures =
	'https://spb.salon-love-forever.ru/upload/resize_cache/iblock/d65/120_180_2/Leda-midi.jpg';
const correctPriceAndStockField = '10';
const correctNameAndDescription = 'платье';

describe('CreateProductForm', () => {
	let addProduct: any;

	beforeEach(() => {
		addProduct = jest.fn();

		jest.spyOn(appStore, 'selectUser').mockReturnValue({
			isLoading: false,
			error: null,
			currentUser: {
				name: 'Александр Филимонов',
				about: 'Frontend10-student',
				avatar:
					'https://images.unsplash.com/photo-1552944150-6dd1180e5999?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=725&q=80',
				_id: '123',
				email: 'somaEmail.av@ya.ru',
				group: 'group-10',
				__v: 0,
			},
		});

		jest
			.spyOn(productsApi, 'useAddProductMutation')
			.mockReturnValue([addProduct, {} as any]);

		testProvidersFn('/users/me', true);
	});

	it('должна рендериться на странице личного кабинета пользователя', () => {
		expect(
			screen.getByTestId(CREATE_PRODUCT_FORM_TEST_ID_MAP.form)
		).toBeInTheDocument();
	});

	describe('поля формы', () => {
		it('должно рендериться поле имени продукта', () => {
			expect(
				screen.getByTestId(CREATE_PRODUCT_FORM_TEST_ID_MAP.nameField)
			).toBeInTheDocument();
		});

		it('должно рендериться поле цены продукта', () => {
			expect(
				screen.getByTestId(CREATE_PRODUCT_FORM_TEST_ID_MAP.priceField)
			).toBeInTheDocument();
		});

		it('должно рендериться поле скидки на продукт', () => {
			expect(
				screen.getByTestId(CREATE_PRODUCT_FORM_TEST_ID_MAP.discountField)
			).toBeInTheDocument();
		});

		it('должно рендериться поле изображений продукта', () => {
			expect(
				screen.getByTestId(CREATE_PRODUCT_FORM_TEST_ID_MAP.picturesField)
			).toBeInTheDocument();
		});

		it('должно рендериться поле кол-ва продукта в наличии', () => {
			expect(
				screen.getByTestId(CREATE_PRODUCT_FORM_TEST_ID_MAP.stockField)
			).toBeInTheDocument();
		});

		it('должно рендериться поле описания продукта', () => {
			expect(
				screen.getByTestId(CREATE_PRODUCT_FORM_TEST_ID_MAP.descriptionField)
			).toBeInTheDocument();
		});
	});

	it('должна рендериться кнопка добавления продукта', () => {
		expect(
			screen.getByTestId(CREATE_PRODUCT_FORM_TEST_ID_MAP.submitButton)
		).toBeInTheDocument();
	});

	describe('работа формы', () => {
		it('не должен добавляться продукт при незаполненном названии продукта', () => {
			const priceField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.priceField
			);
			const picturesField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.picturesField
			);
			const stockField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.stockField
			);
			const descriptionField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.descriptionField
			);
			const submitButton = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.submitButton
			);

			userEvent.type(priceField, correctPriceAndStockField);
			userEvent.type(picturesField, correctPictures);
			userEvent.type(stockField, correctPriceAndStockField);
			userEvent.type(descriptionField, correctNameAndDescription);
			userEvent.click(submitButton);

			waitFor(() => expect(addProduct).not.toHaveBeenCalled());
		});

		it('не должен добавляться продукт при незаполненном поле цены продукта', () => {
			const nameField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.nameField
			);
			const picturesField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.picturesField
			);
			const stockField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.stockField
			);
			const descriptionField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.descriptionField
			);
			const submitButton = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.submitButton
			);

			userEvent.type(nameField, correctNameAndDescription);
			userEvent.type(picturesField, correctPictures);
			userEvent.type(stockField, correctPriceAndStockField);
			userEvent.type(descriptionField, correctNameAndDescription);
			userEvent.click(submitButton);

			waitFor(() => expect(addProduct).not.toHaveBeenCalled());
		});

		it('не должен добавляться продукт при некрректно заполненном поле цены продукта', () => {
			const nameField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.nameField
			);
			const picturesField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.picturesField
			);
			const stockField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.stockField
			);
			const descriptionField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.descriptionField
			);
			const priceField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.priceField
			);
			const submitButton = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.submitButton
			);

			userEvent.type(nameField, correctNameAndDescription);
			userEvent.type(priceField, 'aaa');
			userEvent.type(picturesField, correctPictures);
			userEvent.type(stockField, correctPriceAndStockField);
			userEvent.type(descriptionField, correctNameAndDescription);
			userEvent.click(submitButton);

			waitFor(() => expect(addProduct).not.toHaveBeenCalled());
		});

		it('не должен добавляться продукт при незаполненном поле кол-ва продукта в наличии', () => {
			const nameField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.nameField
			);
			const picturesField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.picturesField
			);
			const priceField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.priceField
			);
			const descriptionField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.descriptionField
			);
			const submitButton = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.submitButton
			);

			userEvent.type(nameField, correctNameAndDescription);
			userEvent.type(picturesField, correctPictures);
			userEvent.type(priceField, correctPriceAndStockField);
			userEvent.type(descriptionField, correctNameAndDescription);
			userEvent.click(submitButton);

			waitFor(() => expect(addProduct).not.toHaveBeenCalled());
		});

		it('не должен добавляться продукт при некорректно заполненном поле кол-ва продукта в наличии', () => {
			const nameField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.nameField
			);
			const picturesField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.picturesField
			);
			const priceField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.priceField
			);
			const descriptionField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.descriptionField
			);
			const stockField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.stockField
			);
			const submitButton = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.submitButton
			);

			userEvent.type(nameField, correctNameAndDescription);
			userEvent.type(stockField, 'aaa');
			userEvent.type(picturesField, correctPictures);
			userEvent.type(priceField, correctPriceAndStockField);
			userEvent.type(descriptionField, correctNameAndDescription);
			userEvent.click(submitButton);

			waitFor(() => expect(addProduct).not.toHaveBeenCalled());
		});

		it('не должен добавляться продукт при незаполненном поле ссылки на изображение продукта', () => {
			const nameField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.nameField
			);
			const stockField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.stockField
			);
			const priceField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.priceField
			);
			const descriptionField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.descriptionField
			);
			const submitButton = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.submitButton
			);

			userEvent.type(nameField, correctNameAndDescription);
			userEvent.type(stockField, correctPriceAndStockField);
			userEvent.type(priceField, correctPriceAndStockField);
			userEvent.type(descriptionField, correctNameAndDescription);
			userEvent.click(submitButton);

			waitFor(() => expect(addProduct).not.toHaveBeenCalled());
		});

		it('не должен добавляться продукт при некорректно заполненном поле ссылки на изображение продукта', () => {
			const nameField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.nameField
			);
			const stockField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.stockField
			);
			const priceField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.priceField
			);
			const descriptionField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.descriptionField
			);
			const picturesField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.picturesField
			);
			const submitButton = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.submitButton
			);

			userEvent.type(nameField, correctNameAndDescription);
			userEvent.type(stockField, correctPriceAndStockField);
			userEvent.type(priceField, correctPriceAndStockField);
			userEvent.type(descriptionField, correctNameAndDescription);
			userEvent.type(picturesField, '123');
			userEvent.click(submitButton);

			waitFor(() => expect(addProduct).not.toHaveBeenCalled());
		});

		it('не должен добавляться продукт при незаполненном описании продукта', () => {
			const nameField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.nameField
			);
			const stockField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.stockField
			);
			const priceField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.priceField
			);
			const picturesField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.picturesField
			);
			const submitButton = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.submitButton
			);

			userEvent.type(nameField, correctNameAndDescription);
			userEvent.type(stockField, correctPriceAndStockField);
			userEvent.type(priceField, correctPriceAndStockField);
			userEvent.type(picturesField, correctPictures);
			userEvent.click(submitButton);

			waitFor(() => expect(addProduct).not.toHaveBeenCalled());
		});

		it('должен добавляться продукт при корректно заполненной форме', () => {
			const nameField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.nameField
			);
			const stockField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.stockField
			);
			const priceField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.priceField
			);
			const descriptionField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.descriptionField
			);
			const picturesField = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.picturesField
			);
			const submitButton = screen.getByTestId(
				CREATE_PRODUCT_FORM_TEST_ID_MAP.submitButton
			);

			userEvent.type(nameField, correctNameAndDescription);
			userEvent.type(stockField, correctPriceAndStockField);
			userEvent.type(priceField, correctPriceAndStockField);
			userEvent.type(descriptionField, correctNameAndDescription);
			userEvent.type(picturesField, correctPictures);
			userEvent.click(submitButton);

			waitFor(() => expect(addProduct).toHaveBeenCalled());
		});
	});

	it("должен соответствовать snapshot'у", () => {
		const tree = ReactTestRenderer.create(<CreateProductForm />).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
