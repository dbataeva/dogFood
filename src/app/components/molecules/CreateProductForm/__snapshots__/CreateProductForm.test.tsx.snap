// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProductForm должен соответствовать snapshot'у 1`] = `
<form
  className="MuiBox-root css-1z13ce1"
  data-testid="form"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    data-testid="nameField"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor=":r0:"
      id=":r0:-label"
    >
      Наименование:
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r0:"
        name="name"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            Наименование:
             
            *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    data-testid="priceField"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor=":r1:"
      id=":r1:-label"
    >
      Цена:
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r1:"
        name="price"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            Цена:
             
            *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    data-testid="discountField"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor=":r2:"
      id=":r2:-label"
    >
      Скидка:
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r2:"
        name="discount"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            Скидка:
             
            *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    data-testid="picturesField"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor=":r3:"
      id=":r3:-label"
    >
      Ссылка на изображение:
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r3:"
        name="pictures"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            Ссылка на изображение:
             
            *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    data-testid="stockField"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor=":r4:"
      id=":r4:-label"
    >
      Количество в наличии:
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r4:"
        name="stock"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            Количество в наличии:
             
            *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    data-testid="descriptionField"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor=":r5:"
      id=":r5:-label"
    >
      Описание товара:
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r5:"
        name="description"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            Описание товара:
             
            *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      data-testid="submitButton"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Добавить продукт
    </button>
  </div>
</form>
`;
